### PR TITLE
Fixed FindBugs warnings

### PR DESCRIPTION
**Pull request by liyinan926**
Signed-off-by: Yinan Li <liyinan926@gmail.com>

### CODE REVIEW

**Comment by liyinan926** [comment]
@zliu41 can you review?

File: gobblin-api/src/main/java/gobblin/configuration/SourceState.java [hunk 1]
```diff
@@ -99,7 +98,7 @@ public SourceState(State properties, SourceState previousSourceState, List<WorkU
    * @return {@link SourceState} of the previous job run
    */
   public SourceState getPreviousSourceState() {
-    return new ImmutableSourceState(this.previousSourceState);
+    return new ImmutableSourceState(this.previousSourceState.or(new SourceState()));
```
Comment (zliu41) [inline gobblin-api/src/main/java/gobblin/configuration/SourceState.java]: It seems the original approach of using a static `EMPTY_SOURCE_STATE` is more efficient. Why the change?

Comment (liyinan926) [inline gobblin-api/src/main/java/gobblin/configuration/SourceState.java]: It was actually doing pre-mature objection creation of the `SourceState` class even before the other members are properly initialized, which is not a good practice.

**Comment by zliu41** [comment]
LGTM
